mmcv/runner/base_runner.py

```diff
@@ -339,6 +339,11 @@ def resume(self,
 
         self._epoch = checkpoint['meta']['epoch']
         self._iter = checkpoint['meta']['iter']
+        if self.meta is None:
+            self.meta = dict()
+        self.meta.setdefault('hook_msgs', dict())
+        # load `last_ckpt`, `best_score`, `best_ckpt`, etc. for hook messages
+        self.meta['hook_msgs'].update(checkpoint['meta']['hook_msgs'])
```

Comment:
Does the key `hook_msgs` necessarily exist in `checkpoint['meta']`?

Reply:
The key `hook_msgs` is used for keeping `best_score`, `best_ckpt`, etc. Which may be useful when resume checkpoint

Reply:
I mean this key may not exist in checkpoints of earlier versions.